Everething is done

Albsch57 opened the pull request:
Please code review

Albsch57 committed:
tableView and two viewController

theswiftvlad commented on TestsToDoManager/ListViewController.swift:
```diff
@@ -0,0 +1,99 @@
+extension ListViewController: PresentViewDelegate {
+    func didCreated(new task: Task) {
+        tasks.append(task)
+        tableView.reloadData()
```
Сделайте не reloadData() - хотя это тоже будет работать, более элегантно будет добавить новую строку в таблицу.
tableView.beginUpdates()
tableView.insertRows
tableView.endUpdates()

theswiftvlad commented on TestsToDoManager/ListViewController.swift:
```diff
@@ -0,0 +1,99 @@
+
+extension ListViewController: PresentViewDelegate {
+    func didCreated(new task: Task) {
+        tasks.append(task)
```
Вы добавляете в tasks массив задачу, а numberOfRowsInSection метод у вас возвращает items.count
Поэтому таблица не может понять, что что то изменилось.

theswiftvlad reviewed (commented):
Вот тут причина

Albsch57 committed:
все приложение To Do

Albsch57 commented on TestsToDoManager/PresentViewController.swift:
```diff
@@ -72,11 +74,21 @@ class PresentViewController: UIViewController {
+            delegate?.didCreated(new: newTask)
+            dismiss(animated: true, completion: nil)
+        }
+        
```
проверьте здесь, пожалуйста. тут должно при изменении текста отображаться написанное и уже редактироваться

theswiftvlad commented on TestsToDoManager/PresentViewController.swift:
```diff
@@ -0,0 +1,114 @@
+}
+
+
+class PresentViewController: UIViewController {
```
У вас нигде в файле нету явной прописи задачи которая редактирует в textfields.
Поэтому контроллер всегда пустой открывается.

theswiftvlad commented on TestsToDoManager/PresentViewController.swift:
```diff
@@ -0,0 +1,114 @@
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
```
Тут нужно проверять если Task не nil, то подставить нужные свойства из предустановленного Task.

theswiftvlad reviewed (changes requested):
Все остальное вроде работает, разве что кнопка круга не работает в ячейке. Но то такое уже.

Albsch57 committed:
сделала UITest
пару вопросов: как работать с bool значениями(проверка если true - это галочка выполнено) и еще вопрос как на клавиатуре вызвать удалить символ
благодарю и хорошого дня:-)

theswiftvlad commented on TestsToDoManagerUITests/ToDoView.swift:
```diff
@@ -0,0 +1,74 @@
+    
+    func buttonSaved() {
+        let button = app.buttons["Create task"]
+        XCTAssert(button.exists)
```
XCTAssert проверки лучше писать в файле теста а не здесь.
Этот файл просто нужен для удобного доступа к UI элементом страницы, XCTAssert нужно вынести в UI тест файл.

theswiftvlad commented on TestsToDoManagerUITests/ToDoView.swift:
```diff
@@ -0,0 +1,74 @@
+    
+    func makeSureEmptyVC() {
+        let taskListScreen = app.navigationBars["To Do List"].firstMatch
+        XCTAssertTrue(taskListScreen.exists, "Не удалось вернуться на экран списка задач.")
```
Тоже вынести в UI Test файл.

theswiftvlad commented on TestsToDoManagerUITests/TestsToDoManagerUITests.swift:
```diff
@@ -8,27 +8,172 @@
+    
+    // Проверка редактирования задачи:
+    func testEditTask() throws {
+        toDoView.maketextField(name: "Yes", description: "To go")
```
Старайтесь давать более осмысленные названия для функций. В данном случае кажется что функция создает TextField, что не есть верным.
Название функции должно прямо передавать смысл ее работы.

theswiftvlad commented on TestsToDoManagerUITests/TestsToDoManagerUITests.swift:
```diff
@@ -8,27 +8,172 @@
+        
+        toDoView.buttonSaved()
+        
+        toDoView.task(to: "selected", boundBy: 0)
```
Не совсем понятно, что делает эта функция.
Она ищет ячейку по индексу - Okay )

Но потом она проверяет наличие кнопки, и нажимает на нее.
_Убедиться, что статус задачи изменился на "Выполнено" и она стала перечеркнутой._ - можно убедится что допустим ее картинка поменялась. Или какой то другой признак сделать, более очевидный что задача стала выполненной. НАпример у выполненных задач можно убирать кнопку Done. Тк она уже Done. И это и будет признаком выполненной задачи.

theswiftvlad commented on TestsToDoManagerUITests/TestsToDoManagerUITests.swift:
```diff
@@ -8,27 +8,172 @@
+        
+        toDoView.closeModalController()
+        
+        toDoView.makeSureEmptyVC()    
```
Нейминг функции не совсем отображает то,  что она делает. По сути она просто должна возвращать текущий заголовок в навигационном баре, ее можно сделать даже как вычисляемое свойство.

theswiftvlad reviewed (changes requested):
Вы Молодец или умница.
Вижу, что нашли где то свое решение, но есть места требующие переосмысления)
